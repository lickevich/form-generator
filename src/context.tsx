import React, { useState } from "react";

import defaultConfig from "./data";
import { ContextType } from "./type";
import { ROUTES } from "./constants";
import { isIncludedValue } from "./utils/helpers";

export const initialState = {
  template: {
    title: "",
    items: [],
    buttons: [],
  },
  setTemplate: () => {},
  error: "",
  setError: () => {},
  config: defaultConfig,
  setConfig: () => {},
  blockedRoutesIds: [ROUTES.RESULT.ID],
  setBlockedRoutesIds: () => {},
  isResultAvailable: false,
};

export const Context = React.createContext<ContextType>(initialState);

export const ContextProvider: React.FC = ({ children }) => {
  const [template, setTemplate] = useState<ContextType["template"]>(
    initialState.template
  );
  const [error, setError] = useState<ContextType["error"]>(initialState.error);
  const [config, setConfig] = useState<ContextType["config"]>(
    initialState.config
  );
  const [blockedRoutesIds, setBlockedRoutesIds] = useState<
    ContextType["blockedRoutesIds"]
  >(initialState.blockedRoutesIds);
  const isResultAvailable: boolean = !isIncludedValue(
    blockedRoutesIds,
    ROUTES.RESULT.ID
  );

  return (
    <Context.Provider
      value={{
        template,
        setTemplate,
        error,
        setError,
        config,
        setConfig,
        blockedRoutesIds,
        setBlockedRoutesIds,
        isResultAvailable,
      }}
    >
      {children}
    </Context.Provider>
  );
};

export default ContextProvider;
