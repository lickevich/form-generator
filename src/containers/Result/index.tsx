import React, { useContext } from "react";

import { Context } from "../../context";
import { ContextType } from "../../type";
import Form from "../../components/Form";

const Result: React.FC<React.ReactNode> = () => {
  const { template } = useContext(Context) as ContextType;

  return <Form {...template} />;
};

export default Result;
