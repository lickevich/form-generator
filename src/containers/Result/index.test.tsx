import React from "react";
import { render, cleanup } from "@testing-library/react";

import { ContextProvider } from "../../context";
import Result from "./";

const renderWithContext = (component: JSX.Element) => {
  return {
    ...render(<ContextProvider>{component}</ContextProvider>),
  };
};

afterEach(cleanup);

describe("Result Page", () => {
  it("render the component", () => {
    const component = renderWithContext(<Result />);
    const { asFragment } = component;

    // @ts-ignore
    expect(asFragment(component)).toMatchSnapshot();
  });
});
