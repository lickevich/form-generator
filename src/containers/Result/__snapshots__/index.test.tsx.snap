// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Page render the component 1`] = `
<DocumentFragment>
  <div
    class="sc-dkPtyc jEDgkp"
  >
    <h1
      class="sc-bdvvaa bDsQHB"
    />
  </div>
  <form
    class="sc-gsDJrp hjUSrA"
  />
  <div
    class="sc-eCImvq UFQQy"
  />
</DocumentFragment>
`;
